app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,23 @@
+class SessionsController < ApplicationController
+  before_action :require_user, only: [:destroy]
+
+  def new
+    redirect_to home_path unless !logged_in?
+  end
+  def create
+    user = User.find_by(email: params[:email])
+
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      flash[:success] = "Welcome again #{user.full_name}"
+      redirect_to home_path
+    else
+      flash[:danger] = "Invalid user or password.Please try again"
+      render :new, params

```

Comment:
You are right, reviewing it with binding.pry you get the parameters. But in the form is not filling the fields with the corresponding param. I´ll have to review it, it used to work on rails 4.0.1 but i´ll have to see why is not working. Thanks.